pkg/boot/universalpayload/universalpayload_test.go

```diff
@@ -37,21 +35,14 @@ func mockGetSMBIOSBase() (int64, int64, error) {
 // to make sure there are no issues with the overall process
 func TestLoadKexecMemWithHOBs(t *testing.T) {
 	// mock data
-	name := "./testdata/upl.dtb"
-	fdtLoad, err := getFdtInfo(name, nil)
-	if err != nil {
-		t.Fatalf("Unexpected error: %+v", err)
-	}
-	data, _ := os.ReadFile(name)
-
 	defer func(old func() (int64, int64, error)) { getSMBIOSBase = old }(getSMBIOSBase)
 	getSMBIOSBase = mockGetSMBIOSBase
 
 	defer func(old func() (*acpi.RSDP, error)) { getAcpiRSDP = old }(getAcpiRSDP)
 	getAcpiRSDP = mockGetRSDP
 
-	defer func(old func() (kexec.MemoryMap, error)) { kexecMemoryMapFromSysfsMemmap = old }(kexecMemoryMapFromSysfsMemmap)
-	kexecMemoryMapFromSysfsMemmap = mockKexecMemoryMapFromSysfsMemmap
+	defer func(old func() (kexec.MemoryMap, error)) { kexecMemoryMapFromIOMem = old }(kexecMemoryMapFromIOMem)
```

Comment:
just FYI, it's very hard to imagine this passing the race detector, but maybe it did. Either way, this really needs to be changed. This is not how this should be done.

Reply:
Good question! Thanks for reminding us of the race detector. These functions are also used in pkg/boot/uefi/uefi_test.go, and we follow the same code style. A better way is passing the functions as variables instead of replacing them.  But this change involes lots of functions and tests. Since this PR is tested and ready for merging, can we do the mocking refactor in next PR?

Reply:
@Raysmond could you draft a PR to follow up and address Ron's suggestion, and post the PR here.

I am going to take this PR in now, as there is an array of others PRs depending on this.